pkg/services/provisioning/dashboards/types.go

```diff
@@ -73,10 +73,16 @@ func createDashboardJson(data *simplejson.Json, lastModified time.Time, cfg *Das
 	return dash, nil
 }
 
-func mapV0ToDashboardAsConfig(v0 []*DashboardsAsConfigV0) []*DashboardsAsConfig {
+func mapV0ToDashboardAsConfig(v0 []*DashboardsAsConfigV0) ([]*DashboardsAsConfig, error) {
 	var r []*DashboardsAsConfig
+	seen := make(map[string]bool)
 
 	for _, v := range v0 {
+		if _, ok := seen[v.Name]; ok {
+			return nil, fmt.Errorf("name %s is not unique", v.Name)
```

Comment:
I would prefer quoting the name in the error message, so that it's unambiguous in case the name contains spaces, and also make it clear that it's a _dashboard_ name:
```suggestion
			return nil, fmt.Errorf("dashboard name %q is not unique", v.Name)
```